src/fields_base.cpp

```diff
@@ -8,20 +8,75 @@
 //
 
 #include <boost/http_proto/fields_base.hpp>
+
+#include <boost/http_proto/error.hpp>
 #include <boost/http_proto/field.hpp>
 #include <boost/http_proto/header_limits.hpp>
+#include <boost/http_proto/rfc/detail/rules.hpp>
+#include <boost/http_proto/rfc/token_rule.hpp>
+
+#include <boost/http_proto/detail/config.hpp>
 #include <boost/http_proto/detail/except.hpp>
-#include "detail/copied_strings.hpp"
-#include "detail/move_chars.hpp"
-#include "detail/number_string.hpp"
-#include <boost/url/grammar/ci_string.hpp>
+
 #include <boost/assert.hpp>
 #include <boost/assert/source_location.hpp>
-#include <string>
+
+#include <boost/url/grammar/ci_string.hpp>
+#include <boost/url/grammar/error.hpp>
+#include <boost/url/grammar/parse.hpp>
+#include <boost/url/grammar/token_rule.hpp>
+
+#include "detail/move_chars.hpp"
 
 namespace boost {
 namespace http_proto {
 
+system::result<typename detail::field_name_rule_t::value_type>
```

Comment:
I would just return `result<string_view>`, using the `value_type` is try-hard with no real-world benefit